reframe/frontend/autodetect.py

```diff
@@ -183,17 +183,29 @@ def detect_topology():
             getlogger().debug(
                 f'> found topology file {topo_file!r}; loading...'
             )
-            part.processor._info = _load_info(
-                topo_file, _subschema('#/defs/processor_info')
-            )
-            found_procinfo = True
+            try:
+                part.processor._info = _load_info(
+                    topo_file, _subschema('#/defs/processor_info')
+                )
+                found_procinfo = True
+            except json.decoder.JSONDecodeError:
```

Comment:
```suggestion
            except json.decoder.JSONDecodeError as e:
```

And then we should print the error message from `e`, unless it's multiline and screws the line.

Reply:
The output in verbose will be now:
```
> found topology file '/Users/ekoutsaniti/.reframe/topology/generic-default/processor.json'; loading...
> error while loading the file: Expecting value: line 1 column 1 (char 0)
> could not load '/Users/ekoutsaniti/.reframe/topology/generic-default/processor.json'; ignoring...
> no topology file found; auto-detecting...
```

Reply:
I've put everything in a single `debug()` call.